nb-api/src/main/java/io/nosqlbench/nb/api/nbio/ResolverForNBIOCache.java

```diff
@@ -0,0 +1,321 @@
+                }
+            } catch (IOException e) {
+                logger.error(() -> "Error downloading remote file to cache at " + cachePath + ", retrying...");
+                retries++;
+            }
+        }
+        return success;
+    }
+
+    private boolean verifyChecksum(Path cachePath, URLContent checksum) {
+        try {
+            String localChecksumStr = generateSHA256Checksum(cachePath.toString());
+            Path checksumPath = Path.of(cachePath.toString().substring(0, cachePath.toString().lastIndexOf('.')) + ".sha256");
+            Files.writeString(checksumPath, localChecksumStr);
+            logger.debug(() -> "Generated local checksum and saved to cache at " + checksumPath);
+            String remoteChecksum = new String(checksum.getInputStream().readAllBytes());
+            if (localChecksumStr.equals(remoteChecksum)) {
+                return true;
+            } else {
+                logger.warn(() -> "checksums do not match for " + checksumPath + " and " + checksum);
+                return false;
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * This method is used to download a file from a remote URL and store it in a local cache.
+     * It first creates the cache directory if it doesn't exist.
+     * Then it tries to download the file and if successful, it generates a SHA256 checksum for the downloaded file.
+     * It then compares the generated checksum with the remote checksum.
+     * If the checksums match, it returns the path to the cached file.
+     * If the checksums don't match or if there was an error during the download, it cleans up the cache and throws a RuntimeException.
+     *
+     * @param uri the URI of the remote file to download
+     * @return the Path to the downloaded file in the local cache
+     * @throws RuntimeException if there was an error during the download or if the checksums don't match
+     */
+    private Path pathFromRemoteUrl(URI uri) {
+        Path cachePath = Path.of(cacheDir + uri.getPath());
+        createCacheDir(cachePath);
+        if (!verifyChecksum) {
+            return execute(NBIOResolverConditions.UPDATE_NO_VERIFY, cachePath, uri);
+        }
+        else {
+            return execute(NBIOResolverConditions.UPDATE_AND_VERIFY, cachePath, uri);
+        }
+    }
+
+    private void createCacheDir(Path cachePath) {
+        Path dir = cachePath.getParent();
+        if (!Files.exists(dir)) {
+            try {
+                Files.createDirectories(dir);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private void cleanupCache(Path cachePath) {
+        if (!cachePath.toFile().delete())
+            logger.warn(() -> "Could not delete cached file " + cachePath);
+        Path checksumPath = Path.of(cachePath.toString().substring(0, cachePath.toString().lastIndexOf('.')) + ".sha256");
```

Comment:
Creating the `checksumPath` happens more than once it should be a method.

Reply:
It's somewhat trivial with the other changes but I went ahead and moved it to a method, it will at least facilitate if we ever want to change strategy here.